python/taichi/profiler/profiler.py

```diff
@@ -0,0 +1,162 @@
+                    record.name)
+            self.statistical_results_[record.name].insert_record(
+                record.kernel_time)
+            self.total_time_ms_ += record.kernel_time
+        self.statistical_results_ = {
+            k: v
+            for k, v in sorted(self.statistical_results_.items(),
+                               key=lambda item: item[1],
+                               reverse=True)
+        }
+
+    def print_info(self, mode=count):
+        self.update_records()  # trace records
+        self.count_results()  # statistical results
+        #count mode (default) : print statistical results of all kernel
+        if mode == self.count:
+            print(
+                "========================================================================="
+            )
+            print(_ti_core.arch_name(ti.cfg.arch).upper() + " Profiler(count)")
+            print(
+                "========================================================================="
+            )
+            print(
+                "[      %     total   count |      min       avg       max   ] Kernel name"
+            )
+            for key in self.statistical_results_:
+                result = self.statistical_results_[key]
+                fraction = result.total_time / self.total_time_ms_ * 100.0
+                print(
+                    "[{:6.2f}% {:7.3f} s {:6d}x |{:9.3f} {:9.3f} {:9.3f} ms] {}"
+                    .format(
+                        fraction,
+                        result.total_time / 1000.0,
+                        result.counter,
+                        result.min_time,
+                        result.total_time / result.counter,  # avg_time
+                        result.max_time,
+                        result.name))
+            print(
+                "------------------------------------------------------------------------"
+            )
+            print(
+                "[100.00%] Total kernel execution time: {:7.3f} s   number of records:  {}"
+                .format(self.total_time_ms_, len(self.statistical_results_)))
+            print(
+                "========================================================================="
+            )
+
+        #trace mode : print records of launched kernel
+        if mode == self.trace:
+            print("====================================")
+            print(_ti_core.arch_name(ti.cfg.arch).upper() + " Profiler(trace)")
+            print("====================================")
+            print("[      % |     time    ] Kernel name")
+            for record in self.traced_records_:
+                fraction = record.kernel_time / self.total_time_ms_ * 100.0
+                print("[{:6.2f}% |{:9.3f}  ms] {}".format(
+                    fraction, record.kernel_time, record.name))
+            print("====================================")
+
+
+_ti_profiler = Profiler()
```

Comment:
Not sure if i'm doing it the right way

Reply:
I think it's fine to have a default profiler instance, but it should  be exposed as a function. (E.g. you can have a `get_default_profiler()`).